### PR TITLE
feat(KD-8225): Rights Form V2

### PR DESCRIPTION
**Pull request by subatomicceo**
## Description of this change

> feat(KD-8225): adds experience form templates; 
> chore(KD-8225): abstracts form types;

## Why is this change being made?
- [ ] Chore (non-functional changes)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested? How can the reviewer verify your testing?

Changes were tested locally in Lanyard:
Screenshot 2023-07-05 at 10 04 21 PM
Screenshot 2023-07-05 at 10 03 30 PM

## Related issues

https://ketch-com.atlassian.net/browse/KD-8225

## Checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [x] I have informed stakeholders of my changes.

### CODE REVIEW

File: src/forms.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+ * @values UNSPECIFIED = type is unspecified
+ *          TEXT = text field
+ *          DROPDOWN = dropdown field
+ *          CHECKBOX = checkbox field
+ */
+export enum FormFieldType {
+  UNSPECIFIED = 'unspecified',
+  TEXT = 'text',
+  DROPDOWN = 'dropdown',
+  CHECKBOX = 'checkbox',
```
Comment (subatomicceo) [inline src/forms.ts:68]: Adds checkbox

File: src/forms.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+ */
+export interface CustomRightForm {
+  rightCode: string
+  formTemplateID: string
+}
+
+/**
+ * FormSection: Describes a section of a form, including its title, description and form fields.
+ */
+export interface FormSection {
```
Comment (subatomicceo) [inline src/forms.ts:104]: Adds FormSection and FormTemplate which should be non breaking changes;

File: src/index.ts [hunk 3]
```diff
@@ -679,92 +681,6 @@ export interface JIT {
   extensions?: { [key: string]: string }
 }
 
-/**
```
Comment (subatomicceo) [inline src/index.ts:682]: Abstracted the form types to a separate file for enhanced readability.

File: src/forms.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+/**
+ * FormTemplate: Describes a form template including its id, code, name, title and sections.
+ */
+export interface FormTemplate {
+  code: string
+  id: string
+  name: string
+  sections: FormSection[]
+  title: string
+}
```
Comment (keepforever) [inline src/forms.ts:119]: Typically, all props on interfaces are optional, i.e. `code?: string`.  Might want to keep that for consistency. Also, forces dev to handle `undefined` case by default.

File: src/index.ts [hunk 5]
```diff
@@ -979,6 +895,7 @@ export interface Right {
    * data subject types
    */
   dataSubjectTypeCodes?: string[]
+  canonicalRightCode: string
```
Comment (keepforever) [inline src/index.ts:898]: maybe optional? `?:`

**Review by keepforever — APPROVED** [review]
approved with comments

**Pull request merged**